---
title: "Empirical comparison of reduced representation bisulfite sequencing and Infinium BeadChip reproducibility and coverage of DNA methylation in humans"
date: 2017-01-01
publishDate: 2019-09-08T18:15:28.603596Z
authors: ["Juan J. Carmona", "William P. Accomando", "Alexandra M. Binder", "John N. Hutchinson", "Lorena Pantano", "Benedetta Izzi", "Allan C. Just", "Xihong Lin", "Joel Schwartz", "Pantel S. Vokonas", "Sami S. Amr", "Andrea A. Baccarelli", "Karin B. Michels"]
publication_types: ["2"]
abstract: "We empirically examined the strengths and weaknesses of two human genome-wide DNA methylation platforms: rapid multiplexed reduced representation bisulfite sequencing and Illumina's Infinium BeadChip. Rapid multiplexed reduced representation bisulfite sequencing required less input DNA, offered more flexibility in coverage, and interrogated more CpG loci at a higher regional density. The Infinium covered slightly more protein coding, cancer-associated and mitochondrial-related genes, both platforms covered all known imprinting clusters, and rapid multiplexed reduced representation bisulfite sequencing covered more microRNA genes than the HumanMethylation450, but fewer than the MethylationEPIC. Rapid multiplexed reduced representation bisulfite sequencing did not always interrogate exactly the same CpG loci, but genomic tiling improved overlap between different libraries. Reproducibility of rapid multiplexed reduced representation bisulfite sequencing and concordance between the platforms increased with CpG density. Only rapid multiplexed reduced representation bisulfite sequencing could genotype samples and measure allele-specific methylation, and we confirmed that Infinium measurements are influenced by nearby single-nucleotide polymorphisms. The respective strengths and weaknesses of these two genome-wide DNA methylation platforms need to be considered when conducting human epigenetic studies."
featured: true
publication: "*npj Genomic Medicine*"
tags: ["selected"]
doi: "10.1038/s41525-017-0012-9"
---

